src/service/resource.js

```diff
@@ -359,6 +363,19 @@ export class Resource {
    * transitioned to the "ready for layout" state.
    */
   measure() {
+    // Check if the element is ready to be measured.
+    // Placeholders are special. They are technically "owned" by parent AMP
+    // elements, sized by parents, but laid out independently. This means
+    // that placeholders need to at least wait until the parent element
+    // has been stubbed. We can tell whether the parent has been stubbed
+    // by whether a resource has been attached to it.
+    if (this.isPlaceholder_ &&
+        this.element.parentElement &&
+        startsWith(this.element.parentElement.tagName, 'AMP-') &&
```

Comment:
How else would we test that this could be AMP, but not sure yet? In most of other APIs we test for definitive: "this is certainly AMP element". Here's it's a bit different.